app/assets/javascripts/components/PipelineSampleReads.jsx

```diff
@@ -2,21 +2,54 @@ class PipelineSampleReads extends React.Component {
 
   constructor(props) {
     super(props);
-    this.pipelineOutput = this.props.pipelineOutput;
-    this.sampleInfo = this.props.sampleInfo;
-    this.projectInfo = this.props.projectInfo;
-    this.reportInfo =  (Object.keys(this.props.reportInfo).length > 0) ? this.props.reportInfo : null;
-    this.jobStatistics = this.props.jobStatistics;
-    this.summary_stats = this.props.summary_stats;
+    this.pipelineOutput = props.pipelineOutput;
+    this.csrf = props.csrf;
+    this.rerunPath = props.rerun_path;
+    this.sampleInfo = props.sampleInfo;
+    this.projectInfo = props.projectInfo;
+    this.reportInfo =  (Object.keys(props.reportInfo).length > 0) ? props.reportInfo : null;
+    this.jobStatistics = props.jobStatistics;
+    this.summary_stats = props.summary_stats;
     this.gotoReport = this.gotoReport.bind(this);
-    this.sampleId = this.sampleInfo.id
+    this.sampleId = this.sampleInfo.id;
+    this.pipelineStatus = props.sample_status
+    this.rerunPipeline = this.rerunPipeline.bind(this);
+    this.state = {
+      rerun: false,
+      failureText: 'Sample run failed'
+    }
   }
 
   gotoReport() {
     $('ul.tabs').tabs('select_tab', 'reports');
     PipelineSampleReads.setTab('pipeline_display','reports');
   }
 
+  shouldProcessPipeline(status) {
+    if (status === null || this.state.rerun === true ) {
```

Comment:
where do you set the status? the status would be null for a few minutes after the run started? 

The rerun button should only show when status === 'FAILED' or 'SUCCESS'

Reply:
yea, this actually checks for status === null, which means the sample is either uploading, initalizing or in progress, it is in the else condition that the rerun button is rendered @yunfangjuan

Reply:
this function should be called "pipelineInProgress then